include/zephyr/mgmt/hawkbit.h

```diff
@@ -36,6 +36,30 @@ enum hawkbit_response {
 	HAWKBIT_PROBE_IN_PROGRESS,
 };
 
+/**
+ * @brief Callback to provide the custom data to the hawkBit server.
+ *
+ * @details This callback is used to provide the custom data to the hawkBit server.
+ * The custom data is used to provide the hawkBit server with the device specific
+ * data.
+ *
+ * @param device_id The device ID.
+ * @param buffer The buffer to store the json.
+ * @param buffer_size The size of the buffer.
+ */
+typedef int (*hawkbit_config_device_data_cb_handler_t)(const char *device_id, uint8_t *buffer,
+						  const size_t buffer_size);
+
+/**
+ * @brief Set the custom data callback.
+ * @details This function is used to set the custom data callback.
+ * The callback is used to provide the custom data to the hawkBit server.
+ * @param cb The callback function.
+ * @retval 0 on success.
+ * @retval -EINVAL if the callback is NULL.
+ */
```

Comment:
We usually have an empty line between `brief`, `details`, `param` and `return`/`retval`, but since `return` is already used in `hawkbit_probe`, it should be used here as well for consistency

Reply:
is now changed